lib/client/api_matrix.es6.js

```diff
@@ -53,6 +53,110 @@ foam.CLASS({
   ],
   methods: [
     {
+      name: 'getBrowsers',
+      documentation: `An asynchronous method returns a group of browser
+        objects, the groups are grouped by browser name and versions.`,
+      returns: {
+        typeName: 'group',
+        documentation: `browsers group by browser name and version.`,
+      },
+      code: function() {
+        return this.browserDAO.select(this.mlang.GROUP_BY(
+          this.Browser.BROWSER_NAME,
+          this.mlang.GROUP_BY(
+            this.Browser.BROWSER_VERSION,
+            this.ArraySink.create())
+        )).then((group) => {
+          let browsers = {};
+          let browserNames = group.groupKeys;
+          for (let i = 0; i < browserNames.length; i++) {
```

Comment:
Maybe I'm missing something, but it seems like the grouping above isn't needed. You end up iterating over the whole list of browsers either way, and a single loop that lazily initializes the browsers by name array would probably be easier to follow.